servicetalk-concurrent-test-internal/src/main/java/io/servicetalk/concurrent/test/internal/AwaitUtils.java

```diff
@@ -52,14 +68,37 @@ public static void awaitUninterruptibly(CountDownLatch latch) {
         }
     }
 
+    /**
+     * Await a {@link CountDownLatch} until the latch is counted down or the given time duration expires,
+     * throws unchecked InterruptedException.
+     * @param latch the {@link CountDownLatch} to await.
+     * @param timeout the timeout duration to await for.
+     * @param unit the units applied to {@code timeout}.
+     * @return true if the count reached zero and false if the waiting time elapsed before the count reached zero
+     */
+    public static boolean await(CountDownLatch latch, long timeout, TimeUnit unit) {
+        final long timeoutNanos = NANOSECONDS.convert(timeout, unit);
+
+        try {
+            return latch.await(timeoutNanos, NANOSECONDS);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throwException(e);
+        }
+
+        return false;
```

Comment:
```suggestion
            throwException(e);
            return false;
        }
```
We can move it under the `catch` block just to make it clear.